### PR TITLE
github hook token

### PR DESCRIPTION
**Pull request by alexcouper**
This includes:
- Rename the existing HOOK_TOKEN to be CORE_API_TOKEN since that is what we're actually using it for.
- Compare the hook_token received from github with a hash of the repo url and see that they match.

Once this has been merged (assuming #27 has been merged) all tests pass.

### CODE REVIEW

File: ployst/github/views.py [hunk 1]
```diff
@@ -25,6 +26,19 @@ def receive_hook(request, hook_token):
...
+
+def create_token(repo_url):
+    """
+    Create a token for the given repo.
+
+    This token will be used in the url of requests from Github.
+    We will then compare it with the repo information contained in the body
+    of the request and if they match then it is deemed valid.
+    """
+    return hashlib.md5(repo_url + settings.GITHUB_HOOK_TOKEN_SALT).hexdigest()
```
Comment (txels) [inline ployst/github/views.py:44]: sha1 or sha256 are safer than md5.
Not that this is military grade ;)

File: ployst/settings/dev_al.py [hunk 2]
```diff
@@ -0,0 +1,5 @@
+from .dev import *  # noqa
+
+GITHUB_CORE_API_TOKEN = "ba6d427e-a893-4014-9633-e38c55210b52"
+CELERY_ALWAYS_EAGER = True
+CELERY_EAGER_PROPAGATES_EXCEPTIONS = True
```
Comment (txels) [inline ployst/settings/dev_al.py:5]: should this one be in `dev.py`?

File: ployst/github/test/test_receive_hook.py [hunk 3]
```diff
@@ -9,9 +11,21 @@
...
 
-    def post(self, data):
+    def _get_hook_token(self, data):
+        "Generate a token for the data provided"
+        try:
+            payload = json.loads(data['payload'])
+            url = payload['repository']['url']
+            return views.create_token(url)
+        except (KeyError, ValueError):
+            return ""
```
Comment (txels) [inline ployst/github/test/test_receive_hook.py:21]: not sure why you need try except here. We'll only call from appropriate test cases.

**Comment by txels** [comment]
Yay green!

File: ployst/settings/dev_al.py (hunk 2 above)
Comment (alexcouper) [inline ployst/settings/dev_al.py:5]: Probably worth while.

File: ployst/github/test/test_receive_hook.py (hunk 3 above)
Comment (alexcouper) [inline ployst/github/test/test_receive_hook.py:21]: Yeah, the tests would need a slight refactor to explicitly generate a hook token or not. Probably worth it though.

**Pull request merged**